### PR TITLE
[INTERNAL] Adopt to task- and middlewareRepository refactorings

### PR DESCRIPTION
**Pull request by RandomByte**
Depends on:
- taskRepository refactoring: https://github.com/SAP/ui5-builder/pull/438
- middlewareRepository refactoring: https://github.com/SAP/ui5-server/pull/302

Also implements:
- Lazy index.js requires

### CODE REVIEW

File: lib/projectPreprocessor.js [hunk 1]
```diff
@@ -587,9 +587,12 @@ class ProjectPreprocessor {
 		const taskRepository = require("@ui5/builder").tasks.taskRepository;
 
 		const taskPath = path.join(extension.path, extension.task.path);
```
Comment (matz3) [inline lib/projectPreprocessor.js]: `extension.path` is already absolute, right?

File: test/lib/extensions.js [hunk 2]
```diff
@@ -616,7 +616,7 @@ test("Project with task extension dependency", (t) => {
 	});
 });
 
-test("Project with task extension dependency - task module not found", async (t) => {
+test("Project with task extension dependency - does not throw for invalid task path", async (t) => {
```
Comment (matz3) [inline test/lib/extensions.js]: As already mentioned in the other PR, what about a test when the `require` of the custom task fails? Which error does the user get?

File: lib/projectPreprocessor.js (hunk 1 above)
Comment (RandomByte) [reply lib/projectPreprocessor.js]: Yes

File: test/lib/extensions.js (hunk 2 above)
Comment (RandomByte) [reply test/lib/extensions.js]: No error here. The require now happens when a different module (e.g. MiddlewareManager or *Builder) tries to retrieve the middleware or task. I added additional error handling there and some tests.

**Pull request merged**